src/Network/Kademlia/Tree.hs

```diff
@@ -141,27 +149,27 @@ handleTimeout tree nid = do
     let f _ _ (nodes, cache) = return $ case L.find (idMatches nid . fst) nodes of
             -- Delete a node that exceeded the limit. Don't contact it again
             --   as it is now considered dead
-            Just x@(_, bs) | bs == pingLimit -> (Bucket (L.delete x $ nodes, cache), False)
+            Just x@(_, PingInfo tc _) | tc == pingLimit -> (Bucket (L.delete x $ nodes, cache), False)
```

Comment:
What is sacramental meaning of `timeoutCount` check? With timestamps we can rely solely on timestamps :)